### PR TITLE
[PAPP-32386] threatstream: Support for custom Observable Types

### PR DESCRIPTION
**Pull request by mposluszny-splunk**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [x] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [x] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [x] If this is a new connector or you are adding new actions
    - [x] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [x] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [x] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [x] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [x] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Currently we are validating provided observable types and limiting choice in action.

## What is the new behavior? (OPTIONAL)
- We are no longer validating observable types, and we are suggesting default types in action in description. There are no dropdown options.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: threatstream.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "publisher": "Splunk",
     "type": "threat intel",
     "main_module": "threatstream_connector.py",
-    "app_version": "3.5.0",
+    "app_version": "3.5.1",
```
Comment (bb-splunk) [inline threatstream.json:8]: If we don't have to do it I wouldn't do it, also this is more than just a patch if we were to split hairs

File: threatstream.json [hunk 2]
```diff
@@ -27232,4 +27225,4 @@
             }
         ]
     }
-}
+}
```
Comment (bb-splunk) [inline threatstream.json:27228]: It was good with the `\n`

File: threatstream_connector.py [hunk 3]
```diff
@@ -132,6 +132,9 @@ def __init__(self):
         self._first_run_limit = None
         self._data_dict = {}  # Blank dict to contain data from all API calls
 
+    def _save_action_handler_progress(self):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
```
Comment (bb-splunk) [inline threatstream_connector.py:136]: Just use the `f-string` please :)

File: threatstream.json [hunk 4]
```diff
@@ -8251,17 +8251,10 @@
             "read_only": false,
             "parameters": {
                 "observable_type": {
-                    "description": "Type of observable to import",
+                    "description": "Type of observable to import, default types: [ip, domain, url, email, hash]. Custom types can be provided if they are available on ThreatStream instance",
```
Comment (bbielinski-splunk) [inline threatstream.json:8254]: Not sure if current description is clear enough. Maybe we can improve it a bit:
```suggestion
                    "description": "Type of observable to import, types supported by default: [ip, domain, url, email, hash]. Custom types can also be provided if they are available on ThreatStream instance",
```

**Review by bb-splunk — APPROVED** [review]

**Review by bbielinski-splunk — CHANGES_REQUESTED** [review]
Please check also if the manual_readme_content should be updated.

File: threatstream.json (hunk 2 above)
Comment (mposluszny-splunk) [reply threatstream.json:27228]: It's the case of my autoformatter to remove \n, can it stay that way?

Comment (bb-splunk) [reply threatstream.json:27228]: It theoretically does no harm but it's one of those good practices that pretty much everyone follows, based on unix's definition of a line, better fix your autoformatter

**Review by bbielinski-splunk — APPROVED** [review]

**Comment by mposluszny-splunk** [comment]
threatstream tests has around 93% of pass rate. Errors are due to resources being unavailable. 
Changed action: "import observable" is passing 100% tests.

**Pull request merged**